src/compiler/emitter.ts

```diff
@@ -805,7 +804,7 @@ namespace ts {
         const emitHost: EmitHost = {
             getPrependNodes: memoize(() => [...prependNodes, ownPrependInput]),
             getCanonicalFileName: host.getCanonicalFileName,
-            getCommonSourceDirectory: () => getNormalizedAbsolutePath(buildInfo.bundle!.commonSourceDirectory, buildInfoDirectory),
+            getCommonSourceDirectory: () => getNormalizedAbsolutePath(buildInfo!.bundle!.commonSourceDirectory, buildInfoDirectory),

```

Comment:
Is the `!` required?  It looks like we checked for undefined above.

Reply:
(And below)

Reply:
arrows dont retain the narrowed values..